src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ArrayOfEmbeddedDataNode.cs

```diff
@@ -23,12 +23,14 @@ public class ArrayOfEmbeddedDataNode<TEmbedded> : ObjectNode
         private ObjectAndOffsetSymbolNode _startSymbol;
         private ObjectAndOffsetSymbolNode _endSymbol;
         private IComparer<TEmbedded> _sorter;
+        private EmbeddedObjectNode _endOfArraySentinel;
```

Comment:
Another example of this would be runs of interface dispatch cells. Currently we just terminate each with the interface method slot number.  The runtime supports series of them, however.  I can subclass ArrayOfEmbeddedDataNode with a TerminatedArrayOfEmbeddedDataNode for these kinds of patterns if you think that's cleaner.